updatecli/updatecli.d/jenkins-inbound-agent.yml

```diff
@@ -24,8 +24,8 @@ sources:
       username: "{{ .github.username }}"
       versionfilter:
         kind: regex
-        ## not the JDK8 version - regex from https://stackoverflow.com/questions/16398471/regex-for-string-not-ending-with-given-suffix/16398502#16398502
-        pattern: '.*[^-][^j][^d][^k][^8]$'
+        ## not the JDK7 version - regex from https://stackoverflow.com/questions/16397471/regex-for-string-not-ending-with-given-suffix/16397502#16397502
+        pattern: '.*[^-][^j][^d][^k][^7]$'
```

Comment:
Not sure what this regexp is supposed to exclude.

Reply:
```suggestion
        pattern: '.*[^-][^j][^d][^k][^8]$'
```

Bad search and replace

Reply:
> Not sure what this regexp is supposed to exclude.

Answer is there: https://github.com/jenkins-infra/docker-hashicorp-tools/pull/183
It used to be a protection until all `-jdk8` images were superseded by new ones (as the GH releases were done in parallel, the "latest" could have been.
The goal of this regex is to exclude releases which have a suffix in `-jdk8` such as https://github.com/jenkinsci/docker-inbound-agent/releases/tag/3028.va_a_436db_35078-2-jdk8 .

![Capture d’écran 2023-10-23 à 11 25 26](https://github.com/jenkins-infra/docker-hashicorp-tools/assets/1522731/6c7cb65e-a748-4a0d-8fd3-ea9d0f578f85)


I believe it is not needed anymore and it should be a subject of another PR to "clean it up"

Reply:
Thanks for the link and explanation, this regexp should definitively be revised in another PR, it doesn't do what it's supposed to do, ex `jdk18` is wrongly excluded:

<img width="357" alt="image" src="https://github.com/jenkins-infra/docker-hashicorp-tools/assets/91831478/bac09f94-b5d9-44cb-a6bd-ae1a34c0e5ae">